### ISSUE TITLE
Unable to install

### ISSUE THREAD

**Issue by jesspritchard** [filamentgroup/grunticon#308]
Hi All,

I've been trying to install/setup grunticon for the past 1.5hrs without much luck. Attached is the npm-debug.log file. It looks to be an issue with imagemin?

I have tried:

* installing in existing web directory by itself (removed existing node_modules folder)
* installing in a brand new empty directory by itself
* installing other dependencies and then running npm install grunt-grunticon --save-dev

None of the above work. In most cases it will freeze and not do anything for a good 15-20min. Imagemin installs normally though.

Node version is v0.10.24 on Mac OS X (10.10.5).

I am about to try and update Node (never done this before) but I thought I'd post already in case it was obvious to anyone.

Thanks,
Jess

[npm-debug.log.zip](https://github.com/filamentgroup/grunticon/files/95740/npm-debug.log.zip)

**Comment by jesspritchard** [comment]
Updating to:

node - v4.2.4
npm - v2.14.12

Fixed these issues however now I get the invalid name: '@filamentgroup/worker-farm' issue. I did read the other issues but didn't understand how to resolve the issue. Is anyone able to explain it to someone with barely any experience in command line/node?

Sidenote: if I run `sudo npm install -g npm@latest` it kills my npm and I get `sudo: npm: command not found`. If I run the installer package again it fixes that issue... am I upgrading it wrong?

**Comment by jefflembeck** [comment]
@jesspritchard have you tried running `npm install -g npm@latest` without `sudo`? That can lead to some problems from time to time.

**Comment by jefflembeck** [comment]
@jesspritchard that being said, this _should_ work for users of npm 2.14 just fine.

Can you do me the favor of going into a new directory and running these three commands?

`node --version`
`npm --version`
`npm install @filamentgroup/worker-farm`

**Comment by jesspritchard** [comment]
@jefflembeck 

npm install -g npm@latest withouth sudo I get the following:

MacBook-Pro:www jesspritchard$ npm install -g npm@latest
npm ERR! Darwin 14.5.0
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "install" "-g" "npm@latest"
npm ERR! node v4.2.4
npm ERR! npm  v2.14.12
npm ERR! path /usr/local/lib/node_modules/npm
npm ERR! code EACCES
npm ERR! errno -13
npm ERR! syscall rmdir

npm ERR! Error: EACCES: permission denied, rmdir '/usr/local/lib/node_modules/npm'
npm ERR!     at Error (native)
npm ERR!  { [Error: EACCES: permission denied, rmdir '/usr/local/lib/node_modules/npm']
npm ERR!   errno: -13,
npm ERR!   code: 'EACCES',
npm ERR!   syscall: 'rmdir',
npm ERR!   path: '/usr/local/lib/node_modules/npm' }
npm ERR! 
npm ERR! Please try running this command again as root/Administrator.
npm ERR! error rolling back Error: EACCES: permission denied, rmdir '/usr/local/lib/node_modules/npm'
npm ERR! error rolling back     at Error (native)
npm ERR! error rolling back  { [Error: EACCES: permission denied, rmdir '/usr/local/lib/node_modules/npm']
npm ERR! error rolling back   errno: -13,
npm ERR! error rolling back   code: 'EACCES',
npm ERR! error rolling back   syscall: 'rmdir',
npm ERR! error rolling back   path: '/usr/local/lib/node_modules/npm' }

Running `node --version` results in  `v4.2.4`

Now running everytime I try to run `npm install -g npm@latest` it seems to break my npm so when I run `npm --version` I get  `-bash: /usr/local/bin/npm: No such file or directory` however after running the node update package again I get  `2.14.12`

Running `npm install @filamentgroup/worker-farm` I get

npm ERR! Darwin 14.5.0
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "install" "@filamentgroup/worker-farm"
npm ERR! node v4.2.4
npm ERR! npm  v2.14.12

npm ERR! Invalid name: "@filamentgroup/worker-farm"
npm ERR! 
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>

Thanks for taking a look - I've read through a lot of the other issues and everything suggested I either don't understand or didn't work.

**Comment by jefflembeck** [comment]
It looks like your issue at hand might be with your node/npm installation :/

That's not great news.

Now, can you tell me what happens when you run:

`sudo npm --version`?

**Comment by jesspritchard** [comment]
@jefflembeck 

`sudo npm --version` outputs `2.14.12`

I was able to install grunticon on my Windows PC at the office. It has node 4.2.3 and npm 2.14.7

When updating node on my Macbook I used the package installer from https://nodejs.org/en/. What would you suggest to try and fix my installation? I'm very much a beginner when it comes to all this and don't have the best knowledge of OSX either :(

**Comment by jefflembeck** [comment]
@jesspritchard Yeah, ok, we have a permissions issue then. This is fixable!

Try going through this: https://docs.npmjs.com/getting-started/fixing-npm-permissions

**Comment by jefflembeck** [comment]
And thanks for putting up with this for us. I really hope we can help you out.

**Comment by jesspritchard** [comment]
@jefflembeck 

I've gone through the Option 1 steps. Is there a way to test if this fixed things?

I've tried running installing grunt-grunticon and @filamentgroup/worker-farm and still get the issues?

Thanks

**Comment by jesspritchard** [comment]
Now that I've had a chance to sit down and try again (rather than trying to do it between projects) I have got it to work.

In case other people have the same problems this is what I did:

As @jefflembeck said, it did appear it was a permissions issue. I followed the instructions in the link @jefflembeck provided (https://docs.npmjs.com/getting-started/fixing-npm-permissions). After this I was able to successfully run `npm install -g npm@latest`. npm is now version 3.5.3

Running `npm install grunt-grunticon --save-dev` then successfully worked.

@jefflembeck Thanks for your help! Much appreciated.
